commons/src/main/java/com/powsybl/commons/extensions/AbstractExtensionAdder.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.extensions;
+
+/**
+ * A base class for implementations of {@link ExtensionAdder} that holds the
+ * extendable to be able build and then add the extension to the extendable.
+ * This class calls {@link Extendable#createExtension} that must be overriden by
+ * subclasses to create the extension.
+ *
+ * @author Jon Harper <jon.harper at rte-france.com>
+ */
+//TODO can we use T extends Extendable<T> here ? Then we can remove the cast in add()
+public abstract class AbstractExtensionAdder<T, E extends Extension<T>> implements ExtensionAdder<T, E> {
+
+    protected T extendable;
+
+    protected AbstractExtensionAdder(T extendable) {
+        this.extendable = extendable;
```

Comment:
2 questions: do we allow null values and is extendable final?

Reply:
There is no problem with forbidding null values (this is always called by doing extendable.newExtensionAdder(..)) so the extendable is never null.
We can also make the extendable final, even private if we want (and pass it to the createExtension method.

You think we should do these changes ?

Reply:
I think it's cleaner if you declare it `private`, `final` and if you assert the extendable is not null